crates/sui-graphql-client/src/lib.rs

```diff
@@ -0,0 +1,568 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+#![doc = include_str!("../README.md")]
+
+pub mod graphql_types;
+
+use base64ct::Encoding;
+use graphql_types::{
+    ChainIdentifierQuery, CheckpointArgs, CheckpointId, CheckpointQuery, CoinMetadata,
+    CoinMetadataArgs, EpochSummaryArgs, EpochSummaryQuery, EventFilter, ObjectFilter, ObjectQuery,
+    ObjectQueryArgs, PageInfo, ProtocolConfigQuery, ProtocolConfigs, ProtocolVersionArgs,
+    ServiceConfig, ServiceConfigQuery, TransactionBlockArgs, TransactionBlockQuery, Uint53,
+};
+use sui_types::types::{
+    Address, CheckpointSequenceNumber, CheckpointSummary, Event, Object, SignedTransaction,
+};
+
+use anyhow::{ensure, Error, Result};
```

Comment:
As a first pass at implementing this library using anyhow is a great choice! Once we've more fleshed things out we should look at using a custom error type to allow for a bit more flexibility/context. But thats something that we should punt till later

Reply:
Sounds good!